### PR TITLE
Refactor filters and tags to work without objectstore

### PR DESCRIPTION
**Pull request by jatindersingh93**
Replace Refresh Button
Make sure data-table updates when object permissions is updated using permission box
Download works on select one and multiple files
Delete works on select one and multiple files
Upload buttons works
Select all works
Detail box works
Remove any orphan codes
[
SC3518](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-3518)

# Description

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: frontend/src/components/object/ObjectFilters.vue [hunk 1]
```diff
@@ -137,13 +130,13 @@ const selectedFilterValuesChanged = () => {
 
 const searchMetadata = async () => {
   searching.value = true;
-  await metadataStore.searchMetadata();
+  await metadataStore.searchMetadata(props['bucketId']);
```
Comment (kyle1morel) [inline frontend/src/components/object/ObjectFilters.vue:133]: `props.bucketId`

File: frontend/src/components/object/ObjectFilters.vue [hunk 2]
```diff
@@ -137,13 +130,13 @@ const selectedFilterValuesChanged = () => {
...
   searching.value = false;
 };
 
 const searchTagging = async () => {
   searching.value = true;
-  await tagStore.searchTagging();
+  await tagStore.searchTagging(props['bucketId']);
```
Comment (kyle1morel) [inline frontend/src/components/object/ObjectFilters.vue:139]: `props.bucketId`

File: frontend/src/store/metadataStore.ts [hunk 3]
```diff
@@ -87,10 +87,10 @@ export const useMetadataStore = defineStore('metadata', () => {
     }
   }
 
-  async function searchMetadata(metadataSet: Array<MetadataPair> = []) {
+  async function searchMetadata(bucketId: string = '', metadataSet: Array<MetadataPair> = []) {
```
Comment (jujaga) [inline frontend/src/store/metadataStore.ts:90]: bucketId should be the second argument, and probably should not hard default to empty string when unspecified, as that will limit this store action's utility in the long run in the event you need to do a COMS-wide search.

File: frontend/src/store/tagStore.ts [hunk 4]
```diff
@@ -82,11 +82,11 @@ export const useTagStore = defineStore('tag', () => {
     }
   }
 
-  async function searchTagging(tagset: Array<Tag> = []) {
+  async function searchTagging(bucketId: string = '', tagset: Array<Tag> = []) {
```
Comment (jujaga) [inline frontend/src/store/tagStore.ts:85]: bucketId should be the second argument, and probably should not hard default to empty string when unspecified, as that will limit this store action's utility in the long run in the event you need to do a COMS-wide search.

File: frontend/src/components/object/ObjectTable.vue [hunk 5]
```diff
@@ -117,7 +119,6 @@ const loadLazyData = (event?: any) => {
       tableData.value = r.data;
       totalRecords.value = +r?.headers['x-total-rows'];
       loading.value = false;
-
       // add to object store
       r.data.forEach((o: COMSObject) => {
         objectStore.$patch((state) => {
```
Comment (kyle1morel) [inline frontend/src/components/object/ObjectTable.vue:124]: Not a change from your PR, but we should avoid using globals like this and instead turn this into a store action.

File: frontend/src/components/object/ObjectTable.vue [hunk 6]
```diff
@@ -173,34 +188,53 @@ const selectedFilters = (payload: any) => {
...
+      @select-all-change="
+        (e) => {
+          selectAll = e.checked;
+          if (e.checked) {
+            selectCurrentPage();
+          } else {
+            objectStore.setSelectedObjects([]);
+          }
+        }
+      "
```
Comment (kyle1morel) [inline frontend/src/components/object/ObjectTable.vue:237]: As Tim mentioned we may be able to remove this logic as we are now always selecting the entire payload.

**Review by TimCsaky — APPROVED** [review]
there are some unused imports but otherwise looks good to me.

**Review by jujaga — APPROVED** [review]

**Pull request merged**